### PR TITLE
[SECURESIGN-298] Add uninstall & template for default values

### PR DESCRIPTION
**Pull request by sallyom**
closed https://github.com/securesign/sigstore-ocp/pull/104 in favor of this PR - see #104 for previous review

This PR:
1) adds @JasonPowr 's `remove keycloak` commit from https://github.com/securesign/sigstore-ocp/pull/102
2) introduces template for default values that is embedded in binary with go compile
3) adds helm lister to check for existence of release to either install or upgrade
4) adds `uninstall` cmd that 
     a) performs helm uninstall
     b) removes extra namespaces that were created but not managed by helm
5) in absence of a custom values file, the OIDC issuer endpoint `https://oauth2.sigstore.dev/auth` will be configured. This is set in the default template/values file. 
6) adds constants for secret file names, certs dir

### CODE REVIEW

**Review by JasonPowr — COMMENTED** [review]
Works Great :) , installed, uninstalled, signed and verified, just two small comments :)

**Comment by JasonPowr** [comment]
I do like the use of go templates to set the host values its much cleaner than my method, I'm assuming something similar could be done for the OIDC provider?

**Comment by osmman** [comment]
I unable to install Helm chart (0.1.25) of TAS via installer because of existing sigstore-monitoring namespace. This namespace does not exist before execution of tas-install. 

`Namespace "sigstore-monitoring" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "trusted-artifact-signer"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "trusted-artifact-signer"`

File: tas-installer/cmd/root.go [hunk 1]
```diff
@@ -23,8 +37,7 @@ var rootCmd = &cobra.Command{
 	For a successful installation, you must have provide the path to a kubeconfig file, or have 
 	one in $HOME/.kube/config. Additionally, the following CLI tools must all be in your $PATH environment.
 	
-	oc - used to install Keycloak
-	openssl - used to create the root certificates	`,
+	oc - used to install Keycloak `,
```
[the remark this answers is not in the document]
Comment (sallyom) [reply tas-installer/cmd/root.go:40]: yup!

File: tas-installer/cmd/uninstall.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+func init() {
+	rootCmd.AddCommand(uninstallCmd)
+}
+
+func uninstallTas() error {
+	log.Print("uninstalling helm chart")
+	msg, err := uninstall.HandleHelmChartUninstall(tasNamespace, tasReleaseName)
+	if err != nil {
+		log.Print(err.Error())
```
[the remark this answers is not in the document]
Comment (sallyom) [reply tas-installer/cmd/uninstall.go:30]: i'm not sure, wdyt? I decided to clear the namespaces regardless of whether the uninstall ends cleanly - that was my thinking.

File: tas-installer/pkg/helm/helm.go [hunk 3]
```diff
@@ -1,62 +1,114 @@
...
-
-	if exists {
-		upgradeRelease(actionConfig, client, settings, chartUrl, chartVersion, values)
-	} else {
-		installNewRelease(actionConfig, client, settings, chartUrl, chartVersion, values)
+	exists := false
+	for _, rel := range releases {
+		if rel.Name == tasReleaseName && rel.Namespace == tasNamespace {
+			exists = true
+			upgradeRelease(actionConfig, client, settings, tasNamespace, chartUrl, chartVersion, values)
```
[the remark this answers is not in the document]
Comment (sallyom) [reply tas-installer/pkg/helm/helm.go:90]: weird, i didn't notice those errors were not being propagated - updating that now

**Comment by Gregory-Pereira** [comment]
> I unable to install Helm chart (0.1.25) of TAS via installer because of existing sigstore-monitoring namespace. This namespace does not exist before execution of tas-install.
> 
Did you run the easy install script first? It would seem to me that [this line](https://github.com/securesign/sigstore-ocp/blob/add-uninstall/tas-easy-install.sh#L105) ran, creating it without the annotations for it being managed by helm. If that is the case you can just delete the project and try again from the insatller. This issue is related to how close some of this work is to the other active PRs in this repo, there are other PRs to handle abstraction of pull-secrets and cert generation.

File: charts/trusted-artifact-signer/templates/cosign-deployment.yaml [hunk 4]
```diff
@@ -30,7 +30,7 @@ spec:
         - name: FULCIO_URL
           value: "https://fulcio.$(OPENSHIFT_APPS_SUBDOMAIN)"
         - name: OIDC_ISSUER_URL
-          value: "https://keycloak-keycloak-system.$(OPENSHIFT_APPS_SUBDOMAIN)/auth/realms/$(OIDC_AUTHENTICATION_REALM)"
+          value: "https://oauth2.sigstore.dev/auth"
```
[the remark this answers is not in the document]
Comment (sallyom) [reply charts/trusted-artifact-signer/templates/cosign-deployment.yaml:33]: sgtm, is this change in another PR? :eyes:

**Comment by sallyom** [comment]
this should be made against the release-1.0.gamma branch, closing

**Pull request closed without merging**